build_extra/rust/BUILD.gn

```diff
@@ -8,1145 +8,1786 @@
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/aesni-x86_64-macosx.S",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/chacha-x86_64-macosx.S",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/ghash-x86_64-macosx.S",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/p256-x86_64-asm-macosx.S",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/poly1305-x86_64-macosx.S",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/sha256-x86_64-macosx.S",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/sha512-x86_64-macosx.S",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/vpaes-x86_64-macosx.S",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/x86_64-mont-macosx.S",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/x86_64-mont5-macosx.S",
+    ]
+  }
+  if (is_win) {
+    libs = [
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/aes-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/aesni-gcm-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/aesni-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/chacha-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/ghash-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/p256-x86_64-asm-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/poly1305-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/sha256-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/sha512-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/vpaes-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/x86_64-mont-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/x86_64-mont5-nasm.obj",
+    ]
+
+    # Suppress "warning: '_addcarry_u64' is not a recognized builtin."
+    cflags = [ "-Wno-ignored-pragma-intrinsic" ]
+  }
+}
+
+## Override: don't build 'ring-test' static library.
+# static_library("ring-test") {
+#   include_dirs = [ "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/include" ]
+#   sources = [ "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/crypto/constant_time_test.c" ]
+# }
```

Comment:
What's this?

Reply:
https://github.com/denoland/deno_third_party/blob/69028eefdc6534683eb1263be0115df679dcaed1/rust_crates/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/build.rs#L429-L448